src/app/app-routing.module.ts

```diff
@@ -0,0 +1,37 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import {AppResolver} from './app-resolver';
+import {NavigationTreeResolver} from './navigation-tree-resolver';
+import {MainWrapperComponent} from './main-wrapper/main-wrapper.component';
+
+const routes: Routes = [
+  { path: '',  redirectTo: '/category?key=1907548051', pathMatch: 'full' },
+  {
+    path: 'category',
+    component: MainWrapperComponent,
+    resolve: {
+      categories: NavigationTreeResolver,
+      products: AppResolver,
+      selectedCategory: AppResolver
+    },
+    runGuardsAndResolvers: 'paramsOrQueryParamsChange',
```

Comment:
You also can replace this route with the following in order to make only one API request to get the list of categories.
```typescript
{
    path: 'category',
    resolve: {
      categories: NavigationTreeResolver
    },
    children: [
      {
        path: '',
        component: MainWrapperComponent,
        resolve: {
          products: AppResolver,
        },
        runGuardsAndResolvers: 'paramsOrQueryParamsChange'
      }
    ]
  }
```